src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1830,11 +1857,57 @@ public static void setInterDCStreamThroughputOutboundMegabitsPerSec(int value)
         conf.inter_dc_stream_throughput_outbound_megabits_per_sec = value;
     }
 
-    public static String[] getAllDataFileLocations()
+    /**
+     * Checks if the local system data must be stored in a specific location which supports redundancy.
+     *
+     * @return {@code true} if the local system keyspaces data must be stored in a different location,
+     * {@code false} otherwise.
+     */
+    public static boolean useSpecificLocationForSystemData()
+    {
+        return conf.system_data_file_directory != null;
+    }
+
+    /**
+     * Returns the locations where the local system keyspaces data should be stored.
+     *
+     * <p>If the {@code system_data_file_directory} was unspecified, the local system keyspaces data should be stored
+     * in the first data directory. This approach guarantees that the server can tolerate the lost of n - 1 disks.</p>
+     *
+     * @return the locations where should be stored the local system keyspaces data
+     */
+    public static String[] getSystemKeyspacesDataFileLocations()
+    {
+        if (conf.system_data_file_directory != null)
+            return new String[] {conf.system_data_file_directory};
+
+        return conf.data_file_directories.length == 0  ? conf.data_file_directories
+                                                       : new String[] {conf.data_file_directories[0]};
```

Comment:
I do not know how people name their directories so I imagine that if we rely on alphabetical order the place were the system files will end up will be a bit of a random one from the user perspective. It sounded easier to me to control the directory ordering than the directory names as it could be handled at the config level.

Reply:
To me using the first one as they are specified in `data_file_directories` sounds easier to understand than an alphabetical selection. The phrase `Cassandra will store the data of the local system keyspaces the first of the data directories specified by data_file_directories` in the documentation seems pretty clear about how the selection is done. We could also add a note in the documentation for `data_file_directories` mentioning that the data of the local system keyspaces will be stored in the first directory, and perhaps warning about the consequences of changing the order of the specified data directories, if we are concerned about accidental order changes.